xtable-utilities/src/main/java/org/apache/xtable/utilities/RunSync.java

```diff
@@ -115,11 +132,39 @@ public static void main(String[] args) throws IOException {
       return;
     }
 
-    DatasetConfig datasetConfig = new DatasetConfig();
+    if (cmd.hasOption(CONTINUOUS_MODE)) {
+      ScheduledExecutorService executorService = Executors.newScheduledThreadPool(1);
+      long intervalInSeconds = Long.parseLong(cmd.getOptionValue(CONTINUOUS_MODE_INTERVAL, "5"));
+      executorService.scheduleAtFixedRate(
+          () -> {
+            try {
+              runSync(cmd);
+            } catch (IOException ex) {
+              log.error("Sync operation failed", ex);
+            }
+          },
+          0,
+          intervalInSeconds,
+          TimeUnit.SECONDS);
+      while (!Thread.interrupted()) {
```

Comment:
This method clears the interrupted status for the thread, do we need that here ? 
`Thread.currentThread().isInterrupted(false)` doesn't clear the interrupted state after the check.

Reply:
nice catch, I need to check the current thread which has an instance method which does not clear the state